### DISCUSSION TITLE
Prisma Operation's such as create, find returning <any, never> instead of the generated type.

### DISCUSSION THREAD

**Discussion by is-it-ayush** [prisma/prisma#19841] [Q&A]
Hi, It's create t3 project and therefore I'm using the default global prisma client approch within a `db.ts` file which instantiates a prisma client which is later added to trpc's request context (`ctx.prisma`). Here's the code for `db.ts` file.
```js
// ~/server/db.ts
import { PrismaClient } from '@prisma/client';
import { env } from "~/env.mjs";

const globalForPrisma = globalThis as unknown as {
  prisma: PrismaClient | undefined;
};

export const prisma =
  globalForPrisma.prisma ??
  new PrismaClient({
    log:
      env.NODE_ENV === "development" ? ["query", "error", "warn"] : ["error"],
  });

if (env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;
``` 
The types are generated (from `npx prisma generate`) safely at 
```
Prisma schema loaded from prisma/schema.prisma

✔ Generated Prisma Client (4.15.0 | library) to ./node_modules/.pnpm/@prisma+client@4.15.0_prisma@4.15.0/node_modules/@prisma/client in 191ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
and I can import them individually via `import { User } from '@prisma/client'`. The problem is, when performing any operation say, 
```
const newUser = await ctx.prisma.user.create({...})
```
the newUser is of type any for unknown reasons. Hovering over create reveals it returns a `Prisma.Prisma__UserClient<any, never>`. I'm not sure if I've done something wrong so any help is appreciated.  Prisma has always worked for me before, I struggle to understand from past two hours why it's not working right now!

Node: v18.16.0
prisma & prisma/client: 4.15.0

**Comment by ludralph** [comment]
Hi @is-it-ayush 👋 

Can you share a reproducible repository with us so we can investigate this further?

**Reply by is-it-ayush** [reply]
Hey @ludralph, I'm unable to reproduce this as I'm not sure of the cause. I've intialized a new repo and it doesn't seem to be an issue there. It's just the problem in my private project. There's not a huge difference between the new repo and my private repo. In fact, we use the same intial configurations in both `schema.prisma` and `~/server/db.ts` (i've pasted the code above). Let me know if I'm able to provide anything that will help you through debug/verbose logs.

For reference, this is what I see. It's `<any, null>` in the return.
![image](https://github.com/prisma/prisma/assets/36449128/6f1b1d76-6db8-451c-b7e7-f90f39b822e9)

**Reply by is-it-ayush** [reply]
Hey @ludralph, I was able to find out the cause. Here's [a repro](https://github.com/is-it-ayush/repro-prisma-type-gen) and you can visit [this line](https://github.com/is-it-ayush/repro-prisma-type-gen/blob/05b4c726a4be74366a0e22320ad71e982fe7e74c/src/server/api/routers/example.ts#L23) and hover over the variables.. It turns out `Key` and `Keys` is used internally by prisma client and prisma client get's confused when exporting the generated types. Now my question is, Should I file a bug?

**Reply by ludralph** [reply]
Hi @is-it-ayush, Can you please file a bug report using this [template](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml)

**Reply by is-it-ayush** [reply]
Hey @ludralph, Sure! Give me a moment.

**Reply by nurul3101** [reply]
Hi! Did you had a chance to create a bug report?

**Reply by is-it-ayush** [reply]
Hey @nurul3101, Yep! I did. [Here](https://github.com/prisma/prisma/issues/19848).

**Reply by nurul3101** [reply]
Thanks for creating the issue. It looks like it's related to a clash between generated types:
- https://github.com/prisma/prisma/issues/19811
We can continue the conversation in the linked issue.
